File: test files/presentation/menu.go
package main

import "fmt"

const board = "Garden menu"

type Special struct {
	Title string
	Price float64
}

// Describe formats a seasonal entry and its total.
func Describe(item Special, portions int) string {
	total := item.Price * float64(portions)
	return fmt.Sprintf("%s: %.2f", item.Title, total)
}

func main() {
	soup := Special{
		Title: "Garden soup",
		Price: 12.5,
	}
	for _, item := range []Special{soup} {
		fmt.Println(board, Describe(item, 2))
	}
}

File: test files/presentation/menu.rs
const BOARD: &str = "Garden menu";

#[derive(Clone, Copy)]
struct Special {
    title: &'static str,
    price: f64,
}

/// Format a seasonal entry and its total.
fn describe(item: &Special, portions: u32) -> String {
    let total = item.price * f64::from(portions);
    format!("{}: {:.2}", item.title, total)
}

fn main() {
    let soup = Special {
        title: "Garden soup",
        price: 12.5,
    };
    let entries = vec![soup];
    for item in &entries {
        println!("{} | {}", BOARD, describe(item, 2));
    }
}
